x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/type/DataType.java

```diff
@@ -192,7 +192,10 @@ public enum DataType {
      * inside alongside time-series aggregations. These fields are not parsable from the
      * mapping and should be hidden from users.
      */
-    PARTIAL_AGG(builder().esType("partial_agg").unknownSize());
+    PARTIAL_AGG(builder().esType("partial_agg").unknownSize()),
+
+    // NOCOMMIT: if we keep this DataType, what should esType be?
+    CATEGORY(builder().esType("category").unknownSize());
```

Comment:
++ Let's' see if we can avoid introducing a new data type. Adding a new type requires touching lots of things to make all work consistently.